locksmith/src/websub/helpers/renewKey.ts

```diff
@@ -0,0 +1,138 @@
+import { ethers, constants, BigNumber } from 'ethers'
+import { Web3Service } from '@unlock-protocol/unlock-js'
+import networks from '@unlock-protocol/networks'
+
+import { KeyRenewal } from '../../models'
+import KeyPricer from '../../utils/keyPricer'
+import Dispatcher from '../../fulfillment/dispatcher'
+
+interface RenewKeyParams {
+  keyId: number
+  lockAddress: string
+  network: number
+}
+
+interface ShouldRenew {
+  shouldRenew: boolean
+  gasRefund?: string
+  error?: string
+}
+
+// precision base for gas calculations
+const BASE = 1000000
```

Comment:
Can you use a better name so we understand what this is for wthout looking at the code please :)